### ISSUE TITLE
Precompile Templates for Production Deployments?

### ISSUE THREAD

**Issue by filipweidemann** [wrabit/django-cotton#77]
Hey there, first of all: this is an insane project, I can't believe the community finally has a package that acknowledges the *extreme* shortcomings of vanilla DTL & brings all the necessary goodies from frameworks like Vue to Django. 

So, hands down a gamechanger for building full-stack apps with HTMX, Alpine & Django. Thank you for that!

That being said, do you plan to add some sort of precompilation method to run on deployments? I read through the code and it seems like the dynamic replacement inside the templates swaps out all `<c-xyz>` tags for the `{% cotton_component %}` template tag and from there on it's just normal Django templating stuff. 

If cotton would ditch the initial compilation on first component load and instead let it be handled through a management command on deploy then I think that the deployment strategy would greatly benefit from that. This could even be done when building, let's say, a Docker image, so not even on startup. What do you think?

**Comment by wrabit** [comment]
Hi, it's great to see some additional validation for the project. Yes, some sort of cache warm/build command is something I have considered. The caching side needs to be looked at in general. Currently it uses LRU cache for holding compiled component templates so as a first step we will need to move to something more centralised like Django's cache (I think LRU will be worker-scoped so on setups like multi-workers on Gunicorn will be difficult to manage) I would benchmark this on different backends before committing fully. 

I would also like to make the cache backend configurable, defaulting to 'default', but to allow for Cotton to be running on a cache backend of the user's choice and will look to include this in this stage.

In summary - it's on the list..

**Comment by filipweidemann** [comment]
Okay sounds great, I'll close this one but maybe this is helpful information for future users. Thanks for answering!
